src/vs/platform/telemetry/common/telemetryService.ts

```diff
@@ -139,9 +141,13 @@ export class TelemetryService implements ITelemetryService {
 				}
 				// Anoynimize user file paths that do not need to be retained or cleaned up.
 				if (!nodeModulesRegex.test(result[0]) && cleanUpIndexes.every(([x, y]) => result.index < x || result.index >= y)) {
-					updatedStack = updatedStack.slice(0, result.index) + result[0].replace(/./g, 'a') + updatedStack.slice(fileRegex.lastIndex);
+					updatedStack += stack.substring(lastIndex, result.index) + '<user-file-path>';
+					lastIndex = fileRegex.lastIndex;
 				}
 			}
+			if (lastIndex < stack.length - 1) {
```

Comment:
I think this should be `if (lastIndex < stack.length)`, otherwise the last character could be omitted.

Reply:
good point. I think I had `<=` few iterations back